stl/inc/iterator

```diff
@@ -1037,14 +1037,17 @@ struct incrementable_traits<common_iterator<_Iter, _Se>> {
     using difference_type = iter_difference_t<_Iter>;
 };
 
-template <class>
+template <class, class>
 struct _Common_iterator_pointer_type {
     using pointer = void;
 };
 
-template <_Has_member_arrow _Iter>
-struct _Common_iterator_pointer_type<_Iter> {
-    using pointer = decltype(_STD declval<_Iter&>().operator->());
+// clang-format off
+template <class _Iter, class _Se>
+    requires _Has_member_arrow<const common_iterator<_Iter, _Se>&> //
+struct _Common_iterator_pointer_type<_Iter, _Se> {
+    // clang-format on

```

Comment:
Mega-nitpick, not worth resetting testing, no change requested: The empty comment `//` (to force wrapping) is unnecessary in a `clang-format off` region.

Reply:
Argh.